src/views/components/input/file/InputFile.native.js

```diff
@@ -1,49 +1,208 @@
 import React, { Component } from 'react';
-import Expo, { ImagePicker } from 'expo';
-import { View, Image, Button } from 'react-native';
+import axios from 'axios';
+import Expo, { ImagePicker, DocumentPicker } from 'expo';
+import { bool, func } from 'prop-types';
+import dlv from 'dlv';
+import fastXmlParser from 'fast-xml-parser';
+import mime from 'react-native-mime-types';
+import { Box } from '../../../components';
+import InputFileItem from './file-item';
+import InputFileTouchable from './file-touchable';
 
 class InputFile extends Component {
   static defaultProps = {
+    imageOnly: false,
   }
 
   static propTypes = {
+    imageOnly: bool,
+    onChange: func,
   }
   
   state = {
-    image: null,
+    files: [],
   };
 
+  handleComplete = result => {
+    this.setState( prevState => ({
+      files: [
+        ...prevState.files,
+        result,
+      ],
+    }), () => {
+      this.handleSaveToServer;
+    });
+  }
+
+  handleSaveToServer = () => {
+    const { files } = this.state;
+    
+    if ( this.props.onChange ) {
+      this.props.onChange({ target: { value: files } });
+    }
+  }
+
+  uploadFile = result => {
+    const localUri = result.uri;
+    const filename = localUri.split( '/' ).pop();
+    const match = /\.(\w+)$/.exec( filename );
+
+    const type = match ? mime.contentType( match[1] ).split( ';' )[0] : 'file';  
+    const formData = new FormData();
+    const url = 'https://uppych40.channel40.com.au/s3/';
```

Comment:
Can we please move this URL into the config file for the time being? Even if we are soon replacing it with store data.